lib/Hooks/FileListener.php

```diff
@@ -512,4 +535,37 @@ private function resetIgnoreCache(Node $node) : void {
 		}
 		$this->ignoreService->clearCacheForStorage($storageId);
 	}
+
+	/**
+	 * @throws NotFoundException
+	 * @throws InvalidPathException
+	 * @throws Exception
+	 */
+	private function onAccessUpdate(int $storageId, int $rootId): void {
+		$userIds = $this->getUsersWithFileAccess($rootId);
+		$files = $this->storageService->getFilesInMount($storageId, $rootId, [ClusteringFaceClassifier::MODEL_NAME], 0, 0);
+		foreach ($files as $fileInfo) {
+			$node = current($this->rootFolder->getById($fileInfo['fileid'])) ?: null;
+			$ownerId = $node?->getOwner()?->getUID();
+			if ($ownerId === null) {
+				continue;
+			}
+			$detectionsForFile = $this->faceDetectionMapper->findByFileId($fileInfo['fileid']);
+			$userHasDetectionForFile = [];
+			foreach ($detectionsForFile as $detection) {
+				$userHasDetectionForFile[$detection->getUserId()] = true;
+			}
+			foreach ($userIds as $userId) {
+				if ($userId === $ownerId) {
+					continue;
+				}
+				if ($userHasDetectionForFile[$userId] ?? false) {
+					continue;
+				}
+				$this->faceDetectionMapper->copyDetectionsForFileFromUserToUser($fileInfo['fileid'], $ownerId, $userId);
+				$this->jobList->add(ClusterFacesJob::class, ['userId' => $userId]);
```

Comment:
it might be good to have a unique set of userIds from among all the `$files` to schedule only one `ClusterFacesJob` per userId